src/utils.js

```diff
@@ -14,3 +14,63 @@ export function defaults(obj, ...args) {
   });
   return obj;
 }
+
+/**
+ * Replaces all occurrences of the specified text. Returns a new value with the replacements made.
+ * This function supports replacing text with React elements and support replacing values within
+ * nested React elements/arrays.
+ *
+ * @param {string|object|Array} interpolated - The value to replace occurrences of the specified text in.
+ * @param {string|RegExp} pattern - The text or regular expression to search for in the interpolated value.
+ * @param {string|object|Array} replacement - The value to replace occurrences of the specified text with.
+ * @returns {string|object|Array} A new value with the specified text replaced.
+ */
+export function replaceValue(interpolated, pattern, replacement) {
+  switch (typeof interpolated) {
+    case 'string': {
+      const split = interpolated.split(pattern);
+      // Check if interpolated includes pattern
+      //  && if String.prototype.replace wouldn't work because replacement is an object like a React element.
+      if (split.length !== 1 && typeof replacement === 'object') {
+        // Return array w/ the replacement
+
+        // React elements within arrays need a key prop
+        if (!replacement.key) {
+          // eslint-disable-next-line no-param-reassign
+          replacement = {...replacement, key: pattern.toString()};
+        }
+
+        return [split[0], replacement, split[1]].flat();
```

Comment:
I don't fully understand this code segment, will this support multiple interpolations of the same key?
It's more of an edge case, but one we should support.

e.g.
```json
   "Key": "My name is {{name}}, and {{name}} is my name!"
```

Reply:
Good question.  This function doesn't, but there's also no need to.  The `parse` function loops though each key.  If the same key exists twice, the replaceValue function will be called twice for that key. 

[This](https://github.com/Shopify/i18next-shopify/blob/26d90d9f1080e735d90b8e561b4fef4473b04f5f/test/shopify_format.spec.js#L178) and [this](https://github.com/Shopify/i18next-shopify/blob/26d90d9f1080e735d90b8e561b4fef4473b04f5f/test/shopify_format_with_react_i18next.spec.js#L65) test ensure we don't have a regression.

Fwiw, this is the same behavior as `String.prototype.replace`, which also only replaces the first match.